src/app/api/v1/admin/users/[primarySha1]/route.ts

```diff
@@ -117,15 +123,34 @@ export async function PUT(
       for (const action of actions) {
         switch (action) {
           case "subscribe": {
+            const sub = await getSubscriberByFxaUid(subscriber.fxa_uid!);
+            const currentFxAProfile = sub.fxa_profile_json as FxaProfile;
+            const subscriptions = new Set(currentFxAProfile.subscriptions);
+
+            subscriptions.add(MONITOR_PREMIUM_CAPABILITY);
```

Comment:
Does the webhook pass us the FxA profile data? I'm a bit nervous about manually modifying our copy of the data, lest it gets out-of-sync with the source of truth...

Reply:
Yes I am concerned about that too, two thoughts I have:

1. FxA is the source of truth, and we synchronize with it via webhook, JWT, or API. The problem here is that we can't re-sync the JWT without sign-out/in, and the profile API doesn't seem to invalidate the cache reliably. 

2. Signing out and back in will cause the JWT to override what's in our database, so I think we won't stay out of sync for too long even if the webhook fails.

I think with the situation around the profile API our only other option here is making users authenticate again (which we could do). There might be a way to force a refresh of the JWT though so it's worth continuing to investigate.

Reply:
OK yeah, if the Webhook doesn't provide the data and the profile API serves outdated data, then we don't have many options left :sweat_smile: But the override at sign-in does alleviate my concerns somewhat.

Reply:
The Webhook does send us a JWT payload, sorry for the confusion - you may be thinking of the OneRep webhook which only sends IDs and not an actual payload. However the client JWT may be outdated which is the problem :/